### PR TITLE
Make TransactionManager::begin/end lock-free.

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

In some profiling runs the TransactionManager begin/end methods showed up very significantly due to high contention on the lock. To improve that, I made them lock-free by putting all counters in a 128-bit struct that is updated in a CAS-loop.

- [x] Bug-Fix for *devel-branch*

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8958/

File: arangod/Cache/TransactionManager.h [hunk 1]
```diff
@@ -70,14 +70,19 @@ class TransactionManager {
...
+  static_assert(sizeof(Counters) == sizeof(uint64_t), "unexpected size");
 
-  void lock();
-  void unlock();
+  struct alignas(16) Data {
+    Counters counters;
+    uint64_t term;
+  };
+
+  std::atomic<Data> _data;
```
Comment (danielhlarkin) [inline arangod/Cache/TransactionManager.h]: I know this is nit-picky, but could you rename `Data` to `State` and `_data` to `_state`?

File: arangod/Cache/TransactionManager.cpp [hunk 2]
```diff
@@ -33,81 +33,68 @@
...
+  Data newData;
+  do {
+    if (((data.term & static_cast<uint64_t>(1)) > 0) && (data.term > tx->term)) {
+      tx->sensitive = true;
+    }
 
-  if (tx->sensitive && (--_openSensitive == 0)) {
-    _term++;
-  }
+    auto newData = data;
```
Comment (danielhlarkin) [inline arangod/Cache/TransactionManager.cpp]:
```suggestion
    newData = data;
```

Comment (danielhlarkin) [reply arangod/Cache/TransactionManager.cpp]: Our CI caught this one, led to an uninitialized value use.

File: arangod/Cache/TransactionManager.h (hunk 1 above)
Comment (mpoeter) [reply arangod/Cache/TransactionManager.h]: Sure! I am also an advocate for good names, so I actually appreciate comments like this - and `State` is indeed a better name here..! :-)

File: arangod/Cache/TransactionManager.cpp (hunk 2 above)
Comment (mpoeter) [reply arangod/Cache/TransactionManager.cpp]: Ah, I missed that in the last refactoring step. This also explains the test failures... unittests ran successfully now on my machine.

**Comment by danielhlarkin** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/8971/

**Comment by jsteemann** [comment]
Thank you!

**Pull request merged**